common/src/main/java/org/opensearch/sdk/SdkClient.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import java.util.concurrent.CompletionStage;
+
+public class SdkClient {
+    private SdkClient impl;
+    
+    public void setImplementation(SdkClient impl) {
+        this.impl = impl;
+    }
+
+    public CompletionStage<PutCustomResponse> putCustom(PutCustomRequest request) {
+        return impl.putCustom(request);
+    }
+
+    public CompletionStage<GetCustomResponse> getCustom(GetCustomRequest request) {
+        return impl.getCustom(request);
+    }
+
+    public CompletionStage<DeleteCustomResponse> deleteCustom(DeleteCustomRequest request) {
+        return impl.deleteCustom(request);
+    }
```

Comment:
We could but I'm not sure of the value of that.  Nothing requires the implementation to act asynchronously, I could always immediately complete the future (and in fact for some error handling/illegal argument cases, would immediately complete exceptionally).    Sync methods would require more exception throwing/handling.

Not opposed to it if we need it but I don't see the need for it (yet).

Reply:
I think it is primarily with coding style of the plugin. Given that we are building a generic interface it might be better to support both and let client chose their implementation style.

Reply:
Makes sense, but the comment about exception handling remains.  With the `CompletionStage` we clearly can pass exceptions back to the caller, even if executed synchronously.   Should we consistently throw a particular type of exception from these methods, and if so, what type?  I'd lean toward `OpenSearchStatusException` lacking any other preference.

Reply:
So, we can't have two methods with the same name and arguments, but different return type.
```java
    public CompletionStage<PutCustomResponse> putCustom(PutCustomRequest request);

    public PutCustomResponse putCustom(PutCustomRequest request);
```
Possible to add an "async" name to the CompletionStage version.

Also we could just have default "synchronous" implementations:
```java
    public CompletionStage<PutCustomResponse> putCustom(PutCustomRequest request);

    default PutCustomResponse putCustomSync(PutCustomRequest request) {
        CompletableFuture<PutCustomResponse> future = new CompletableFuture<>();
        putCustom(request).whenComplete((r, ex) -> {
            if (r == null) {
               future.completeExceptionally(ex);
            } else {
                future.complete(r);
            }            
        });
        try {
            return future.get();
        } catch (InterruptedException ie) {
            FutureUtils.cancel(future);
            Thread.currentThread().interrupt();
            throw new OpenSearchException(ie);
        } catch (java.util.concurrent.ExecutionException e) {
            throw new OpenSearchException(e.getCause());
        }
    }
```

Reply:
yes, this approach looks good.